docs/Guides/Recommendations.md

```diff
@@ -349,3 +350,29 @@ It is perfectly fine to spin up several Fastify instances within the same
 Node.js process and run them concurrently, even in high load systems. 
 Each Fastify instance only generates as much load as the traffic it receives,
 plus the memory used for that Fastify instance.
+
+## Causes Of Performance Degradation
```

Comment:
How does this constitute a recommendation?

Reply:
This section can be considered recommendations of what not to do to make sure you don't unintentionally worsen performance

Reply:
This content should live in a blog post IMHO

Example: https://backend.cafe/the-secrets-behind-high-performance-with-nodejs

I think so because we can't say that without measuring and if some Node.js version optimizes the RegExp stuff. It would not be true anymore.

Then, let me be realistic:

- a regexp can't be faster than a positional string
- an async function with async I/O, can't be faster than a sync function without I/O
- BUT a sync function can be slower than an async function if I run a loop+crypto logic..
- in general: more business logic is slower than less business logic

While I understand these general rules of thumb, they are confusing and misleading.

We should provide the user the info to choose by themself instead of saying: "avoid that because it is slow".

A good blog post with a benchmark could help people to choose.

Reply:
> This content should live in a blog post IMHO

Couldn't agree more as someone who regularly reads backend.cafe 😁